diff --git a/include/bat/ledger/publisher_info.h b/include/bat/ledger/publisher_info.h
index f4f210c..7c9ddd0 100644
--- a/include/bat/ledger/publisher_info.h
+++ b/include/bat/ledger/publisher_info.h
@@ -36,7 +36,7 @@ LEDGER_EXPORT enum PUBLISHER_MONTH {
 };
 
 LEDGER_EXPORT struct PublisherInfoFilter {
-  PublisherInfoFilter(int category_, 
+  PublisherInfoFilter(int category_,
     PUBLISHER_MONTH month_, const std::string& year_);
 
   int category;
@@ -81,7 +81,7 @@ LEDGER_EXPORT struct PublisherInfo {
   std::string year;
 };
 
-using PublisherInfoList = std::vector<const PublisherInfo>;
+using PublisherInfoList = std::vector<PublisherInfo>;
 
 }  // namespace ledger
 
